test/topology_custom/test_tablets2.py

```diff
@@ -243,6 +243,100 @@ async def check():
 
     await cql.run_async("DROP KEYSPACE test;")
 
+async def get_two_servers_to_move_tablet(manager: ManagerClient):
+    logger.info("Bootstrapping cluster")
+    cmdline = []
+    servers = [await manager.server_add(cmdline=cmdline)]
+
+    await manager.api.disable_tablet_balancing(servers[0].ip_addr)
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE test WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 1};")
+    await cql.run_async("CREATE TABLE test.test (pk int PRIMARY KEY, c int);")
+
+    servers.append(await manager.server_add(cmdline=cmdline))
+
+    key = 7 # Whatever
+    tablet_token = 0 # Doesn't matter since there is one tablet
+    await cql.run_async(f"INSERT INTO test.test (pk, c) VALUES ({key}, 0)")
+    rows = await cql.run_async("SELECT pk from test.test")
+    assert len(list(rows)) == 1
+
+    replica = await get_tablet_replica(manager, servers[0], 'test', 'test', tablet_token)
+    logger.info(f'{replica=}')
+
+    s0_host_id = await manager.get_host_id(servers[0].server_id)
+    s1_host_id = await manager.get_host_id(servers[1].server_id)
+
+    dst_shard = 0
+
+    return (servers, cql, s0_host_id, s1_host_id, replica, tablet_token, dst_shard)
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_streaming_rx_error_no_failed_message_with_fail_stream_plan(manager: ManagerClient):
+    servers, cql, s0_host_id, s1_host_id, replica, tablet_token, dst_shard = await get_two_servers_to_move_tablet(manager)
+
+    await manager.api.enable_injection(servers[0].ip_addr, "stream_session_ignore_failed_message", one_shot=True)
+    await manager.api.enable_injection(servers[1].ip_addr, "stream_session_ignore_failed_message", one_shot=True)
+    await manager.api.enable_injection(servers[1].ip_addr, "stream_mutation_fragments_rx_error", one_shot=True)
```

Comment:
You're assuming that server[1] will be the receiving server (replica[1]), which is not guaranteed.

Reply:
Correct. I updated the test to make sure server[0] is the source.